File: ios/SpentyAI/Core/Models/Invoice.swift
import Foundation

struct LineItem: Codable, Hashable, Identifiable {
    var id: String { "\(description ?? "")-\(quantity)-\(rate)" }

    var description: String?
    var quantity: Double
    var rate: Double
    var discountPercent: Double?
    var gstRate: Double?
    var hsnSac: String?
    var amount: Double?
    var taxAmount: Double?

    enum CodingKeys: String, CodingKey {
        case description
        case quantity
        case rate
        case discountPercent = "discount_percent"
        case gstRate = "gst_rate"
        case hsnSac = "hsn_sac"
        case amount
        case taxAmount = "tax_amount"
    }
}

struct Invoice: Codable, Identifiable, Hashable {
    var id: String { invoiceId }

    let invoiceId: String
    let userId: String?
    var invoiceNumber: String?
    var invoiceType: String?
    var invoiceDate: String?
    var dueDate: String?
    var customerId: String?
    var customerName: String?
    var customerGstin: String?
    var customerAddress: String?
    var customerState: String?
    var placeOfSupply: String?
    var lineItems: [LineItem]?
    var subtotal: Double?
    var taxTotal: Double?
    var grandTotal: Double?
    var paymentStatus: String?
    var amountPaid: Double?
    var paymentAccountId: String?
    var paymentMethod: String?
    var paymentDate: String?
    var poNumber: String?
    var notes: String?
    var termsConditions: String?
    var createdAt: String?

    enum CodingKeys: String, CodingKey {
        case invoiceId = "invoice_id"
        case userId = "user_id"
        case invoiceNumber = "invoice_number"
        case invoiceType = "invoice_type"
        case invoiceDate = "invoice_date"
        case dueDate = "due_date"
        case customerId = "customer_id"
        case customerName = "customer_name"
        case customerGstin = "customer_gstin"
        case customerAddress = "customer_address"
        case customerState = "customer_state"
        case placeOfSupply = "place_of_supply"
        case lineItems = "line_items"
        case subtotal
        case taxTotal = "tax_total"
        case grandTotal = "grand_total"
        case paymentStatus = "payment_status"
        case amountPaid = "amount_paid"
        case paymentAccountId = "payment_account_id"
        case paymentMethod = "payment_method"
        case paymentDate = "payment_date"
        case poNumber = "po_number"
        case notes
        case termsConditions = "terms_conditions"
        case createdAt = "created_at"
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/models/Invoice.kt
package com.spentyai.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Invoice(
    @SerialName("invoice_id") val id: String,
    @SerialName("invoice_number") val invoiceNumber: String,
    @SerialName("customer_id") val customerId: String,
    @SerialName("customer_name") val customerName: String? = null,
    val status: InvoiceStatus = InvoiceStatus.DRAFT,
    @SerialName("issue_date") val issueDate: String,
    @SerialName("due_date") val dueDate: String,
    val subtotal: Double = 0.0,
    @SerialName("tax_amount") val taxAmount: Double = 0.0,
    @SerialName("discount_amount") val discountAmount: Double = 0.0,
    val total: Double = 0.0,
    @SerialName("amount_paid") val amountPaid: Double = 0.0,
    @SerialName("amount_due") val amountDue: Double = 0.0,
    val currency: String = "USD",
    val notes: String? = null,
    val terms: String? = null,
    @SerialName("line_items") val lineItems: List<InvoiceLineItem>? = null,
    @SerialName("payment_date") val paymentDate: String? = null,
    @SerialName("sent_date") val sentDate: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)

@Serializable
enum class InvoiceStatus {
    @SerialName("draft") DRAFT,
    @SerialName("sent") SENT,
    @SerialName("viewed") VIEWED,
    @SerialName("paid") PAID,
    @SerialName("overdue") OVERDUE,
    @SerialName("cancelled") CANCELLED,
    @SerialName("partially_paid") PARTIALLY_PAID
}

@Serializable
data class InvoiceLineItem(
    val description: String,
    val quantity: Double = 1.0,
    @SerialName("unit_price") val unitPrice: Double = 0.0,
    val amount: Double = 0.0,
    @SerialName("tax_rate") val taxRate: Double? = null
)
